### PR TITLE
New Quiz LTI labeling

### PR DESCRIPTION
**Pull request by rh12**
refs: [MBL-17960](https://instructure.atlassian.net/browse/MBL-17960)
affects: Student, Teacher, Parent
release note: Improved support for New Quizzes.

## What's new
New Quizes should look like quizes, see ticket for details.
Also updated at some additional places:
- Modules screen
- Assignments screen
- Grades screen
- ToDo screen
- People Details screen
- PARENT app: Grades tab
- Assignment Details screen
- Submission Details screen (+ Comments sheet)
- Quizzes screen (not displaying untrue "0 Questions" for QuizLTIs anymore)

## What's not changed
- Notifications screen
  - not updated, because `users/self/activity_stream` doesn't provide details
  - we could check via `html_url` which contains `assignmentID`, see `ActivityStreamViewController`
  - not sure it's worth it
- Calendar Events list
  - not updated, because `planner/items` doesn't provide details
  - we could check `plannable_id` when `plannable_type` is `assignment`, see `Plannable.iconImage`
  - not sure it's worth it
- Did not add nice to have panda

## Test plan
- See ticket and above changed / not changed list.
- Test both Teacher and Student apps, they use different screens for similar things.
- Also smoke test for other type of assignments.

## TODO:
- screenshots

--- edit or delete this section ---
## Screenshots

<table>
<tr><th>Before</th><th>After</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist
- [ ] Follow-up e2e test ticket created
- [ ] A11y checked
- [ ] Tested on phone
- [ ] Tested on tablet
- [ ] Tested in dark mode
- [ ] Tested in light mode
- [ ] Approve from product

[MBL-17960]: https://instructure.atlassian.net/browse/MBL-17960?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Parent Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Teacher Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student, Teacher, Parent

[MBL-17960](https://instructure.atlassian.net/browse/MBL-17960)

<details><summary>:x: Swift lint</summary>


    
    ----- xcbeautify -----
    Version: 2.11.0
    ----------------------
    
    ❌ /Users/vagrant/git/Core/CoreTests/Assignments/AssignmentTests.swift:416:1: Trailing Whitespace Violation: Lines should not have trailing whitespace (trailing_whitespace)
    /Users/vagrant/git/Core/CoreTests/Modules/ModuleItemTests.swift:96:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace (trailing_whitespace)
    /Users/vagrant/git/Core/CoreTests/Modules/ModuleItemTests.swift:99:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace (trailing_whitespace)
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0c52e99600b6e4c2f5949e12087ce1e79c94ca86
</p>

**Review by suhaibabsi-inst — COMMENTED** [review]
Thanks @rh12 !

Code +1

With minor notes to consider.

**Review by suhaibabsi-inst — CHANGES_REQUESTED** [review]
Tested on iPhone, I was able to see the new labeling reflected as described in the PR description. Beside button is showing the correct label on all details screens of Quiz.

One issue through, is that questions count label doesn't show on Assignments screen. And it shows count 0 for new quizzes on Quizzes screen.

Simulator Screenshot - iPhone 16 - 2024-12-04 at 17 07 24

Simulator Screenshot - iPhone 16 - 2024-12-04 at 17 07 00

Still would need to test that on iPad as well.

**Comment by rh12** [comment]
> One issue through, is that questions count label doesn't show on Assignments screen. And it shows count 0 for new quizzes on Quizzes screen.

It is not displayed for Classic Quizzes either, so I would consider it a separate feature.
It seems to me the PR about Assignment List doesn't contain this either.
To be honest I'm not sure that this is something we want there, since it's also not listed on the Grades screen as well, which is a kind of "inspiration" for the new Assigmnets screen.
Feel free to bring it up as an improvement though, and discuss with UX/Product.

**Review by vargaat — CHANGES_REQUESTED** [review]

File: Core/Core/Modules/ModuleList/ModuleItemCell.swift
Comment (rh12) [reply Core/Core/Modules/ModuleList/ModuleItemCell.swift:87]: Moving the `type` calculation to a private extension var `displayedType`

File: Core/Core/Modules/ModuleList/ModuleItemCell.swift
Comment (rh12) [reply Core/Core/Modules/ModuleList/ModuleItemCell.swift:141]: Moving the `type` calculation to a private extension var `displayedType`

File: Core/Core/Assignments/AssignmentDetails/AssignmentDetailsView.swift
Comment (rh12) [reply Core/Core/Assignments/AssignmentDetails/AssignmentDetailsView.swift:223]: I would avoid further littering the model type with view related properties. This should probably live in the related ViewModel, but we don't have one here.
As a lesser evil, I'm moving this to a private extension.

File: Core/Core/Database.xcdatamodeld/Database.xcdatamodel/contents
Comment (rh12) [reply Core/Core/Database.xcdatamodeld/Database.xcdatamodel/contents:714]: Removing it, both for `Assignment` and `ModuleItem`

**Review by suhaibabsi-inst — APPROVED** [review]
@rh12 

I got confused with what's shown on the Figma file. If time permits, let's fix the count number issue on Quizzes screen. So, to match with what we have for old quizzes.

Approving so not to block the merge.

**Comment by rh12** [comment]
@suhaibabsi-inst 
> If time permits, let's fix the count number issue on Quizzes screen. So, to match with what we have for old quizzes.

Sorry, I overlooked this one in your previous comment. You're right, zero is not good here.
It turns out backend doesn't send a `question_count` for QuizLTIs, it's also not visible on the web.
But the web hides the Question count text altogether in this case, so I did the same.

**Review by vargaat — CHANGES_REQUESTED** [review]
Looks like we have a CoreData validation error when downloading modules.
IMG_0427

**Comment by rh12** [comment]
@vargaat 
> Looks like we have a CoreData validation error when downloading modules.

Thanks for the fix!
QA+1 on this on my side, too

**Review by suhaibabsi-inst — DISMISSED** [review]
QA +1

**Review by vargaat — DISMISSED** [review]
QA+1

**Review by ndrsszsz — APPROVED** [review]

**Review by suhaibabsi-inst — APPROVED** [review]
QA +1

**Pull request merged**
